File: captainhcg/go/1-two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for idx, v := range(nums) {
        complement := target - v
        i, ok := m[complement]
        if ok {
            return []int{i, idx}
        }
        m[v] = idx
    }
    return []int{0, 0}  // it should never execute
}

File: captainhcg/js/1-two-sum.js
var twoSum = function(nums, target) {
    var map = new Map();
    for(var idx = 0; idx < nums.length; idx += 1){
        var complement = target - nums[idx];
        if(map.has(complement)){
            return [map.get(complement), idx]
        }
        map.set(nums[idx], idx);
    }
};
